multiqc/modules/picard/HsMetrics.py

```diff
@@ -346,9 +335,26 @@ def _generate_table_header_config(table_cols, hidden_table_cols):
             for s, r in title_cleanup:
                 h_title = h_title.replace(s, r)
 
+            # Extract the coverage from the column name
+            m = re.match(r".+_(\d+)X", h)
+            if m:
+                try:
+                    cov = int(m.group(1))
+                except ValueError:
```

Comment:
Given that the regex pattern is `\d+` is this not guaranteed to always work?

Reply:
Good point! Simplified the code